dcm4che-core/src/main/java/org/dcm4che3/data/BulkData.java

```diff
@@ -463,12 +444,17 @@ public int hashCode() {
 
     /** Returns the index after the segment ends */
     public long getSegmentEnd() {
-        return offset() + longLength();
+    	if( length==-1 ) return -1;
+        return offset() + (length & 0xFFFFFFFFl);
     }
 
     /** Gets the actual length as a long so it can represent the 2 gb to 4 gb range of lengths */
     public long longLength() {
-        return length & 0xFFFFFFFFl;
+        return length;
     }
 
+    @FunctionalInterface
```

Comment:
Do you use this new Creator interface anywhere?
I don't really get the purpose.

Reply:
For some reason the cherry pick added this. I don't see it used anywhere either so I will remove it.